core/chain_manager.go

```diff
@@ -616,14 +615,12 @@ func (self *ChainManager) InsertChain(chain types.Blocks) (int, error) {
 		stats      struct{ queued, processed, ignored int }
 		tstart     = time.Now()
 
-		nonceDone    = make(chan nonceResult, len(chain))
-		nonceQuit    = make(chan struct{})
 		nonceChecked = make([]bool, len(chain))
 	)
 
 	// Start the parallel nonce verifier.
-	go verifyNonces(self.pow, chain, nonceQuit, nonceDone)
-	defer close(nonceQuit)
+	nonceAbort, nonceResults := verifyBlockNonces(self.pow, chain)
+	defer close(nonceAbort)
```

Comment:
No, imagine the following scenario:

Verifier: add task 1 to queue
Worker: fetch tast 1 from queue
Inserter: close verification quit channel
Inserter: close results channel
Verifier: close tasks channel
Worker: Finish verifying task 1, write into results channel
panic